src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/MvcDataAnnotationsMvcOptionsSetup.cs

```diff
@@ -13,33 +13,43 @@ namespace Microsoft.AspNetCore.Mvc.DataAnnotations.Internal
     /// </summary>
     public class MvcDataAnnotationsMvcOptionsSetup : IConfigureOptions<MvcOptions>
     {
-        private IServiceProvider _serviceProvider;
-
-        public MvcDataAnnotationsMvcOptionsSetup(IServiceProvider serviceProvider)
+        private IStringLocalizerFactory _stringLocalizerFactory;
+        private IValidationAttributeAdapterProvider _validationAttributeAdapterProvider;
+        private IOptions<MvcDataAnnotationsLocalizationOptions> _dataAnnotationLocalizationOptions;
+
+        public MvcDataAnnotationsMvcOptionsSetup(
```

Comment:
You'd need two constructors here since `IStringLocalizerFactory` is optional, one with and one without `IStringLocalizerFactory`.